### DISCUSSION TITLE
How avoid loss precision when retrive data from sql server

### DISCUSSION THREAD

**Discussion by vanthuc96** [sqlalchemy/sqlalchemy#8649] [Usage Questions]
I'm using sqlalchemy to create connection with my database (sql server), and I realized that when I try to query data, some columns is saved as datetime (ex: '2022-08-22 11:18:00.000') and when I retrieve data, it seem data lost a bit (ex: '2022-08-22 11:18:00')
I just execute a very simple sql command.
. I'm newbie in python and it's first time I work on this library. Does anyone has any solutions to avoid this loss ?Thank you for reading

**Comment by CaselIT** [comment]
Hi,

does this happen only when the ms are 000?

**Reply by zzzeek** [reply]
I would check on fractional seconds settings for the datatype in use.

**Reply by vanthuc96** [reply]
@CaselIT yes, when ms has other values, it will be return a microsecond
